userspace/falco/atomic_signal_handler.h

```diff
@@ -85,23 +85,21 @@ class atomic_signal_handler {
 	inline bool handle(std::function<void()> f) {
 		if(triggered() && !handled()) {
 			std::unique_lock<std::mutex> lock(m_mtx);
-			if(!handled()) {
```

Comment:
I think is worth keeping this check since we are checking an atomic value that might be changed in between calls.

Reply:
In that case, shouldn't `m_mtx` rather get lock at the start of the function ? Or just check for `handle` after the lock took place ?